src/Microsoft.Health.Dicom.Functions.Client/DicomAzureFunctionsClient.cs

```diff
@@ -239,6 +239,23 @@ public async Task StartContentLengthBackFillOperationAsync(Guid operationId, Can
         _logger.LogInformation("Successfully started content length backfill operation with ID '{InstanceId}'.", instanceId);
     }
 
+    /// <inheritdoc/>
+    public async Task<OperationReference> StartDeleteExtendedQueryTagOperationAsync(Guid operationId, string tagPath, CancellationToken cancellationToken = default)
+    {
+        EnsureArg.IsNotNull(tagPath, nameof(tagPath));
+
+        cancellationToken.ThrowIfCancellationRequested();
+
+        string instanceId = await _durableClient.StartNewAsync(
```

Comment:
Changed to get the tag & update the status before the operation is called. If the tag is not there or the status is already set to deleting, it will throw an exception

Reply:
One thing I'm not sure about is what to do if the entire operation fails. It is OK that the state will be left in 'Deleting'? 

That means if they resubmit the request afterwards, it will always fail because the state is `Deleting`. That might be ok, since if the first operation had failing then probably a resubmission would fail as well, but lets say we put in a bug fix to fix whatever made the operation fail. Then we would also have to manually fix the tag's status.